### DISCUSSION TITLE
Sun altitude

### DISCUSSION THREAD

**Discussion by elidamd** [UMEP-dev/UMEP#532] [General]
Hi,

I need some clarification to understand the sun's altitude.
I ran a simulation on Envimet and imported the output data to QGIS, x,y UTM coordinates.
Then I ran a simulation with Solweig, using the same forcing data values, same day (21.06.2015), and hours (5 am to 21 pm).
Below I list the zenith angle given for each simulated hour of Envimet and the sun altitude from UMEP.

I understand that sun's altitude is complementary to the zenith. But here these values don't match.

hour Altitude Umep Zenith Envimet
5 3,89 7,41961
6 11,49 15,54098
7 19,9 24,21889
8 28,72 33,11556
9 37,57 41,83378
10 46 49,81206
11 53,31 56,16081
12 58,37 59,59227
13 59,88 59,01337
14 57,28 54,63500
15 51,46 47,72084
16 43,75 39,46609
17 35,14 30,65191
18 26,26 21,78011
19 17,52 13,22346
20 9,29 5,31478
21 2,04 0,00000

How is the sun altitude calculated, please?
If I am using the same date/hour and UTM coordinates, what else could be diverting the results?

Thank you!

**Comment by CsillaVG** [comment]
Hi Élida,

Some models use the middle of the hour altitude in their hourly insolation calculation. You can compare these values to NOAA's Solar Position Calculator and see which model is working with the hourly values and which one with the middle of the hour ones.

Best,
Csilla

**Reply by elidamd** [reply]
Dear Csilla,

Thanks a lot for your contribution! I compared both sun ranges with NOAA's calculator and found out that Umep is based on the middle of the hours while Envimet uses the full hour, but in their case, the summertime is not considered. 
I wonder if the calculations in Solweig results for PET are based on the sun's position in the middle hours,  or if they essentially consider the radiation inputs, which refer to full-hour measurements.
Do you know about it?
Thanks!
Best,
Élida

**Reply by biglimp** [reply]
Hi Élida, the PET calculations comes from SOLWEIG so it make use of the same time, i.e. middle of time step. /Fredrik

**Reply by elidamd** [reply]
Hi Fredrik, 
Thanks for the quick reply! Is there a way to retrieve results based on full-hours? Does Solweig use the radiation inputs, or does it calculate new values that fit to the middle-hours?

**Reply by biglimp** [reply]
Without having tested it, I think you can make a small adjustment in the code below by setting halftimestepdec = 0 in the file Solweig_v2015_metdata_nolaod.py to calculate sun position on the hour.

https://github.com/UMEP-dev/UMEP/blob/493da7d033f3117a0e8575fe4c3b1fc009eaf1bf/Utilities/SEBESOLWEIGCommonFiles/Solweig_v2015_metdata_noload.py#L27

You need to restart QGIS to make the changes active.

**Reply by elidamd** [reply]
Thank you! I will test and post here the findings afterward.
